### PR TITLE
fixed tls issues

### PR DESCRIPTION
**Pull request by rajatagrawal-dev**
#6891 
#6895

### CODE REVIEW

File: Tasks/Common/TlsHelper_/TlsHelper_.psm1 [hunk 1]
```diff
@@ -5,16 +5,20 @@ function Add-Tls12InSession {
     param()
 
     try {
-        if ([Net.ServicePointManager]::SecurityProtocol -notcontains 'Tls12') {
-            [Net.ServicePointManager]::SecurityProtocol += [Net.SecurityProtocolType]3072
+        if ([Net.ServicePointManager]::SecurityProtocol.ToString().Split(',').Trim() -notcontains 'Tls12') {
```
Comment (arjgupta) [inline Tasks/Common/TlsHelper_/TlsHelper_.psm1]: this should be tostring().Trim().split(',')

Comment (Ajay-MS) [reply Tasks/Common/TlsHelper_/TlsHelper_.psm1]: Split(',').Trim() - this will trim invidual values. I think this is fine here.

Comment (asranja) [inline Tasks/Common/TlsHelper_/TlsHelper_.psm1]: Should we use the HasFlag() method of Enum class? The if condition would look like if ( -Not [System.Net.ServicePointManager]::SecurityProtocol.HasFlag([System.Net.SecurityProtocolType]::Tls12) )

Refer: https://msdn.microsoft.com/en-us/library/system.enum.hasflag(v=vs.110).aspx

Comment (Ajay-MS) [reply Tasks/Common/TlsHelper_/TlsHelper_.psm1]: +1

Comment (rajatagrawal-dev) [reply Tasks/Common/TlsHelper_/TlsHelper_.psm1]: hasFlag is available since .Net 4.0 only. Let's not take that dependency.

Comment (rajatagrawal-dev) [reply Tasks/Common/TlsHelper_/TlsHelper_.psm1]: Yes, Ajay is right. This should be fine.

File: Tasks/Common/TlsHelper_/TlsHelper_.psm1 [hunk 2]
```diff
@@ -5,16 +5,20 @@ function Add-Tls12InSession {
...
+            $securityProtocol=@()
+            $securityProtocol+=[Net.ServicePointManager]::SecurityProtocol
+            $securityProtocol+=[Net.SecurityProtocolType]3072
+            [Net.ServicePointManager]::SecurityProtocol=$securityProtocol
+            
             Write-Host (Get-VstsLocString -Key TLS12AddedInSession)
```
Comment (asranja) [inline Tasks/Common/TlsHelper_/TlsHelper_.psm1]: Do we need to expose this as a user facing message ?

Comment (rajatagrawal-dev) [reply Tasks/Common/TlsHelper_/TlsHelper_.psm1]: Yes. We'll let the user know that we have added TLS 1.2 in session and that the network calls might use this protocol.

**Comment by jessehouwing** [comment]
Shouldn't this have been part of the PowerShell execution handler?  Or at least part of the task-lib?

**Pull request merged**